lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -807,6 +809,16 @@ is_unnecessary_quote([Part], #elixir_tokenizer{warn_on_unnecessary_quotes=true}
 is_unnecessary_quote(_Parts, _Scope) ->
   false.
 
+unsafe_to_atom(Part, Line, Column, #elixir_tokenizer{static_atoms_encoder=StaticAtomsEncoder} = Scope) when
+    is_function(StaticAtomsEncoder) ->
+  Metadata = [{line, Line}, {column, Column}, {file, Scope#elixir_tokenizer.file}],
+  Value = elixir_utils:characters_to_binary(Part),
+  case StaticAtomsEncoder(Value, Metadata) of
+    {ok, Term} ->
+      {ok, Term};
+    {error, Reason} ->
+      {error, {Line, Column, Reason, Part}}
+  end;
```

Comment:
Does having this clause before the next one mean that `:statis_atoms_encoder` will let atoms that are bigger than 255 codepoints work? Since the tokenizer now raises an error for atoms that go over the system limit, maybe it's better to keep that behaviour?

Reply:
I actually did this on purpose because when you pass an atom encoder, I think it should be up to yourself to check limits like these. I can imagine an encoder function that takes too long atom names and transforms them into atoms that fit the system limit.

Reply:
@arjan I am still not sure about this. I think it depends on what we want this functionality to address. If we want this to let people parse random code safely, then I would go with keeping the hard limit on 255 characters.

Reply:
I'm fine with changing this behaviour around so that any atoms-to-be are first length-checked. It's an edge case anyway.

Reply:
I think 255 is sane limit to have anyway, so let's move the limit check clause up.

Reply:
Fixed in 970ba29